File: software/examples/php/ExampleThreshold.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletOzone.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletOzone;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change XYZ to the UID of your Ozone Bricklet

// Callback function for ozone concentration reached callback (parameter has unit ppb)
function cb_ozoneConcentrationReached($ozone_concentration)
{
    echo "Ozone Concentration: $ozone_concentration ppb\n";
}

$ipcon = new IPConnection(); // Create IP connection
$o = new BrickletOzone(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Get threshold callbacks with a debounce time of 10 seconds (10000ms)
$o->setDebouncePeriod(10000);

// Register ozone concentration reached callback to function cb_ozoneConcentrationReached
$o->registerCallback(BrickletOzone::CALLBACK_OZONE_CONCENTRATION_REACHED,
                     'cb_ozoneConcentrationReached');

// Configure threshold for ozone concentration "greater than 20 ppb" (unit is ppb)
$o->setOzoneConcentrationCallbackThreshold('>', 20, 0);

echo "Press ctrl+c to exit\n";
$ipcon->dispatchCallbacks(-1); // Dispatch callbacks forever

?>

File: software/examples/ruby/example_threshold.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_ozone'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change XYZ to the UID of your Ozone Bricklet

ipcon = IPConnection.new # Create IP connection
o = BrickletOzone.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Get threshold callbacks with a debounce time of 10 seconds (10000ms)
o.set_debounce_period 10000

# Register ozone concentration reached callback
o.register_callback(BrickletOzone::CALLBACK_OZONE_CONCENTRATION_REACHED) do |ozone_concentration|
  puts "Ozone Concentration: #{ozone_concentration} ppb"
end

# Configure threshold for ozone concentration "greater than 20 ppb"
o.set_ozone_concentration_callback_threshold '>', 20, 0

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect
